components/switch/demo/switch.html

```diff
@@ -0,0 +1,54 @@
+<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<meta name="viewport" content="width=device-width, initial-scale=1.0">
+		<meta http-equiv="X-UA-Compatible" content="ie=edge">
+		<meta charset="UTF-8">
+		<link rel="stylesheet" href="../../demo/styles.css" type="text/css">
+		<script src="/node_modules/@webcomponents/webcomponentsjs/webcomponents-loader.js"></script>
+		<script type="module">
+			import '../../demo/demo-page.js';
+			import '../../tooltip/tooltip.js';
+			import '../switch.js';
+		</script>
+	</head>
+	<body unresolved>
+
+		<d2l-demo-page page-title="d2l-switch">
+
+			<h2>On/Off</h2>
+
+			<d2l-demo-snippet>
+				<template>
+					<d2l-switch label="Tool" on></d2l-switch>
+				</template>
+			</d2l-demo-snippet>
+
+			<h2>On/Off (hidden label)</h2>
+
+			<d2l-demo-snippet>
+				<template>
+					<d2l-switch label="Tool" label-hidden></d2l-switch>
+				</template>
+			</d2l-demo-snippet>
+
+			<h2>Visibility</h2>
+
+			<d2l-demo-snippet>
+				<template>
+					<d2l-switch label="Visibility" on-icon="tier1:visibility-show" off-icon="tier1:visibility-hide"></d2l-switch>
```

Comment:
I think the "visibility" use-case may be fairly common.  In the current implementation it is supported by exposing `on-icon` and `off-icon` property.  However I am considering a couple of other options:

1. expose a `type="visibility"` property and bake the icons into the component for that variant; or
2. factor this logic into a mixin and provide two separate components - `d2l-switch` and `d2l-switch-visibility`

It may be necessary to change the text depending on the state... "Visible" and "Hidden", however we should be careful with this because screen reader (at least VoiceOver announces: `"{label}, {on/off}, switch"`, so given that, changing the label could make the announcement very confusing.

Reply:
Yeah I'd be in favour of exposing this as a separate component seeing as this is such a common use case.

Reply:
Updated to define `d2l-switch` and `d2l-switch-visibility`.